requester/requester_internal_test.go

```diff
@@ -43,7 +48,9 @@ func TestRun(t *testing.T) {
 			label: "return ErrConnection early on connection error",
 			req: New(config.Config{
 				Request: config.Request{
-					Timeout: 0,
+					Method:  "POST",
+					Timeout: 1,
```

Comment:
Ah yes, it's because of the host that the error should indeed be returned.
I don't understand why adding a body changes anything though. I have retried and there is no `ErrConnection` raised with `Timeout: 0` (locally at least). I got one record which is a success.